glotaran/project/project_model_registry.py

```diff
@@ -0,0 +1,45 @@
+"""The glotaran model registry module."""
+from __future__ import annotations
+
+from pathlib import Path
+
+from glotaran.io import load_model
+from glotaran.project.generators.generator import GeneratorArguments
+from glotaran.project.generators.generator import generate_model_yml
+from glotaran.project.project_registry import ProjectRegistry
+
+
+class ProjectModelRegistry(ProjectRegistry):
+    """A registry for models."""
+
+    def __init__(self, directory: Path):
+        """Initialize a model registry.
+
+        Parameters
+        ----------
+        directory : Path
+            The registry directory.
+        """
+        super().__init__(directory / "models", ".yml", load_model)
+
+    def generate_model(
+        self,
+        name: str,
+        generator: str,
+        generator_arguments: GeneratorArguments,
+    ):
+        """Generate a model.
+
+        Parameters
+        ----------
+        name : str
+            The name of the model.
+        generator : str
+            The generator for the model.
+        generator_arguments : GeneratorArguments
+            Arguments for the generator.
+        """
+        model_yml = generate_model_yml(
```

Comment:
Make writing a model file conditional and add an additional parameter `overwrite_existing`
```python
if (self._directory / f"{name}.yml").is_file() is not True or overwrite_existing is True:
```

Reply:
done